x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/InlineStats.java

```diff
@@ -121,15 +122,18 @@ public boolean expressionsResolved() {
     public List<Attribute> output() {
         if (this.lazyOutput == null) {
             List<NamedExpression> addedFields = new ArrayList<>();
-            AttributeSet childOutput = child().outputSet();
+            AttributeSet set = child().outputSet();
 
             for (NamedExpression agg : aggregates) {
-                if (childOutput.contains(agg) == false) {
+                Attribute att = agg.toAttribute();
+                if (set.contains(att) == false) {
                     addedFields.add(agg);
+                    set.add(att);
```

Comment:
If you're trying to solve the output order in cases like `INLINESTATS .. BY x = ...`, I think we need to amend `ReplaceStatsNestedExpressionWithEval`.

This fella does
```
INLINESTATS .. BY x = ...

->
EVAL x = ...
| INLINESTATS ... BY x
```
But that's wrong because for an existing group `x`, `INLINESTATS` will try to leave the group where it was.

2 Solutions:
1. We change INLINESTATS so that it always places the groups to the far right. That'd be consistent with STATS and will be easy to build with little specialcasing.
2. We fix the optimizer; we could do that by giving `INLINESTATS` a fixed output order that's determined at analysis/resolution time; or we could change `ReplaceStatsNestedExpressionWithEval` so that for INLINESTATS specificallly, it does

```
INLINESTATS .. BY x = ...

->
EVAL x = ...
| INLINESTATS ... BY x
| PROJECT ..., x
```

Reply:
Bleh, upon closer inspection I think I understand what this is trying to achieve; this is for cases like `BY x, x`, right? Let's give it a comment at least before merging?

Reply:
I'm still confused by the changes to `Inlinestats.output` and _think_ they're redundant. But!

I see the problem with

```
row x = 1, y = 2 | inlinestats max(y) by x, x
```

It's that the first phase is optimized to
```
Aggregate[STANDARD,[x{r}#13, x{r}#13],[MAX(y{r}#15) AS max(y), x{r}#13, x{r}#13]]
\_Row[[1[INTEGER] AS x, 2[INTEGER] AS y]]
```
The problems are: duplicate `x` group and wrong output order (we need to reorder the aggregates or add a projection after). In `Inlinestats.firstPhase` we just hand over the aggregates and groupings to an `Aggregate`, but at that point we can still have duplicate groupings and ill-ordered aggregates.

Reply:
Yeah. I can revert this change I think. It was an attempt but it shouldn't stick.